python/jupytergis_lab/jupytergis_lab/notebook/gis_document.py

```diff
@@ -839,6 +883,98 @@ def create_source(
         return None
 
 
+def vector_to_geojson(
+    filepath,
+    out_geojson=None,
+    bbox=None,
+    mask=None,
+    rows=None,
+    epsg="4326",
+    encoding="utf-8",
+    **kwargs,
+):
+    """Converts any geopandas-supported vector dataset to GeoJSON.
+
+    Args:
+        filepath (str): Either the absolute or relative path to the file or URL
+            to be opened, or any object with a read() method (such as an open
+            file or StringIO).
+        out_geojson (str, optional): The file path to the output GeoJSON.
+            Defaults to None.
+        bbox (tuple | GeoDataFrame or GeoSeries | shapely Geometry, optional):
+            Filter features by given bounding box, GeoSeries, GeoDataFrame or
+            a shapely geometry. CRS mis-matches are resolved if given a GeoSeries
+            or GeoDataFrame. Cannot be used with mask. Defaults to None.
+        mask (dict | GeoDataFrame or GeoSeries | shapely Geometry, optional):
+            Filter for features that intersect with the given dict-like geojson
+            geometry, GeoSeries, GeoDataFrame or shapely geometry. CRS mis-matches
+            are resolved if given a GeoSeries or GeoDataFrame. Cannot be used with
+            bbox. Defaults to None.
+        rows (int or slice, optional): Load in specific rows by passing an integer
+            (first n rows) or a slice() object.. Defaults to None.
+        epsg (str, optional): The EPSG number to convert to. Defaults to "4326".
+        encoding (str, optional): The encoding of the input file. Defaults to "utf-8".
+        kwargs: Additional arguments to pass to geopandas.read_file.
+
+
+    Raises:
+        ValueError: When the output file path is invalid.
+
+    Returns:
+        dict: A dictionary containing the GeoJSON.
+    """
+
+    try:
+        import geopandas as gpd
+    except ImportError:
+        raise ImportError(
+            "geopandas is required for this function. Please install it using `pip install geopandas`."
+        )
+
+    if not filepath.startswith("http"):
+        filepath = os.path.abspath(filepath)
+        if filepath.endswith(".zip"):
+            filepath = "zip://" + filepath
+    ext = os.path.splitext(filepath)[1].lower()
+    if ext == ".kml":
+        try:
+            import fiona
+        except ImportError:
+            raise ImportError(
+                "fiona is required for this function. Please install it using `pip install fiona`."
+            )
+
+        fiona.drvsupport.supported_drivers["KML"] = "rw"
+        df = gpd.read_file(
+            filepath,
+            bbox=bbox,
+            mask=mask,
+            rows=rows,
+            driver="KML",
+            encoding=encoding,
+            **kwargs,
+        )
+    else:
+        df = gpd.read_file(
```

Comment:
I guess this will download the GeoJSON if I do something like:
```python
url = "https://github.com/opengeos/datasets/releases/download/world/countries.geojson"
doc.add_vector_layer(path=url, name="GeoJSON")
```

This is unfortunate since it will embed the GeoJSON data into the JGIS file, which may be an unwanted behavior. Keeping the GeoJSON source pointed by URL may be better.

So I guess we'd want to consider two approaches in this function:
- if it's a URL, we keep the URL as-is and don't embed the data
- if it's a local file or some in-memory data in Python, we don't have the choice but to embed the data so we keep this logic here